File: LABORATORIO 7/PREG01_LAB08_2024-2/Biblioteca/DetalleComanda.cpp
//
// Created by USER on 11/06/2026.
//

#include "DetalleComanda.h"

#include "Bebida.h"
#include "Entrada.h"
#include "PlatoFondo.h"

DetalleComanda::DetalleComanda() {
    id=0;
    pedido= nullptr;
}

DetalleComanda:: ~DetalleComanda() {
    if (pedido !=nullptr) delete pedido;

}

int DetalleComanda::get_id() const {
    return id;
}

void DetalleComanda::set_id(int id) {
    this->id = id;
}


// 142,B,Cafe,2,grande
// 142,B,Agua Mineral,2.5,pequeno
// 142,E,Quesadillas,5.5,sin picante

void DetalleComanda::leecomandas(ifstream &arch) {
    char c, tipo;
    arch>>id;
    if (arch.eof()) return;
    arch>>c>>tipo>>c;
    if (tipo== 'B')
        pedido = new Bebida;
    if (tipo == 'E')
        pedido = new Entrada;
    if (tipo == 'P')
        pedido = new PlatoFondo;
    pedido->lee(arch);


}

void DetalleComanda::imprime(ofstream &arch) {
    pedido->imprimir(arch);
    arch<<endl;
}

File: Restaurante_2024-2/src/DetalleComanda.java
import java.util.Scanner;

public class DetalleComanda {
    private int id;
    private Producto pedido;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
//142 B Cafe 2 grande
//142 B Agua_Mineral 2.5 pequeno
//142 E Quesadillas 5.5 sin_picante
    public void  leecomanda(Scanner arch){
        String tipo;
        id=arch.nextInt();
        tipo= arch.next();
        if(tipo.compareTo("B")==0)
            pedido = new Bebida();
        else
            if(tipo.compareTo("E")==0)
                pedido = new Entrada();
            else
                pedido = new PlatoFondo();

        pedido.lee(arch);
    }

    public void imprimecomanda(){
        System.out.println("Id: "+id);
        pedido.imprime();
        System.out.println("Tipo: "+pedido.tiposoy());
    }

}
